src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/DefaultGeneratedDocumentPublisher.cs

```diff
@@ -78,13 +78,14 @@ public override void PublishCSharp(string filePath, SourceText sourceText, long
             {
                 HostDocumentFilePath = filePath,
                 Changes = textChanges,
-                HostDocumentVersion = hostDocumentVersion,
+                HostDocumentVersion = (long)hostDocumentVersion,
```

Comment:
Instead of casting this we should change our UpdateBufferRequest type. It used to expect `long` and now it doesn't